src/python/pants/engine/README.md

```diff
@@ -74,8 +76,43 @@ class FormattedInt(datatype(['content'])): pass
 @rule(FormattedInt, [Select(IntType)])
 def int_to_str(an_int):
   return FormattedInt('{}'.format(an_int))
+
+# Field values can be specified with positional and/or keyword arguments in the constructor:
+x = FormattedInt('a string')
+x = FormattedInt(content='a string')
+
+# Field values can be accessed after construction by name or index:
+print(x.content)    # 'a string'
+print(x[0])         # 'a string'
+
+# datatype objects can be easily inspected:
+print(x)            # 'FormattedInt(content=a string)'
```

Comment:
As a separate change, it looks like we should be using the `repr` of fields, rather than `str`. That would result in quotes around string fields in this case.

Reply:
This is the output of the `__str__()` method on datatypes (which also displays any type constraints on fields) -- the `__repr__()` method uses the `repr` of each field. Does that sound appropriate, or is there a further change to be made?

Reply:
(if `print()` uses `repr`, then i may have taken a shortcut in producing that example code/output -- i really hope not because i hate seeing that)

Reply:
Looks like that's exactly what happened. I can make a quick followup to fix that.

Reply:
See #5818 for small followup.